src/vcpkg/postbuildlint.cpp

```diff
@@ -973,6 +974,99 @@ namespace vcpkg::PostBuildLint
+        }
+        std::vector<std::string> string_paths;
+        for (const auto& path : absolute_paths)
+        {
+            string_paths.push_back(path.native());
+#if defined(_WIN32)
+            string_paths.push_back(path.generic_u8string());
+#endif
+        }
+
+        std::string result;
+        for (const auto& path_and_contents : files_and_contents)
+        {
+            const auto extension = path_and_contents.first.extension().substr(1 /* ignore dot */);
+            const bool is_header = extension == "h" || extension == "hpp" || extension == "hxx";
+            if (Util::any_of(string_paths, [&path_and_contents, extension, is_header](const std::string& path) {
+                    if (extension == "cfg" || extension == "conf")
+                    {
+                        return Strings::contains(path_and_contents.second, path);
+                    }
+                    for (size_t offset = 0;;)
+                    {
+                        const auto index = path_and_contents.second.find(path, offset);
+                        if (index == std::string::npos) return false;
+                        if (is_header)
+                        {
+                            bool new_line = false;
+                            for (auto start = index;;)
+                            {
+                                const auto before = path_and_contents.second.find_last_of("\n/", start);
+                                if (before == std::string::npos) return true;
+                                if (path_and_contents.second[before] == '\n')
+                                {
+                                    if (before == 0) return true;
+                                    new_line = true;
+                                    start = before - 1;
+                                    continue;
+                                }
+                                if (path_and_contents.second[before + 1] == '*') break;              // is in a comment
+                                if (before == 0) return true;                                        // not in a comment
+                                if (!new_line && path_and_contents.second[before - 1] == '/') break; // is in a comment
+                                if (path_and_contents.second[before - 1] == '*') return true; // is not in a comment
+                                start = before - 1;
+                            }
+                        }
+                        else
+                        { // .py, .sh, .cmake or .pc file
+                            const auto before = path_and_contents.second.find_last_of("\n#", index);
+                            if (before == std::string::npos) return true;
+                            if (path_and_contents.second[before] == '\n') return true; // not a comment
+                        }
+                        offset = index + path.size();
+                    }
+                }))
+            {
```

Comment:
given that this is a warning, not a hard error, I'd prefer we just checked for the absolute paths without trying to look for whether we're in a comment. This is just a lot of complexity for little gain, imo.

Reply:
It is a hard error in the ci...

Reply:
right but we can fix those issues.

Reply:
We could, but this is a lot of work for nothing

Reply:
We should extract this and make it unit testable then.